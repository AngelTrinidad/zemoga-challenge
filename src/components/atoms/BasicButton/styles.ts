import {StyleSheet} from 'react-native';

import {scaleSpace} from '~helpers';

const styles = StyleSheet.create({
  button: {
    width: '100%',
    minHeight: scaleSpace(50),
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 12,
  },
});

export default styles;
