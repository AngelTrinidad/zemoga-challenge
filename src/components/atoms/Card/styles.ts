import {StyleSheet} from 'react-native/';

const styles = StyleSheet.create({
  card: {
    borderRadius: 20,
    shadowColor: 'black',
    shadowOpacity: 0.5,
    shadowOffset: {
      width: 0,
      height: 4,
    },
    elevation: 11,
  },
});

export default styles;
