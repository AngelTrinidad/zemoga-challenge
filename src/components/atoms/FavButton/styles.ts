import {StyleSheet} from 'react-native';

import {scaleSpace} from '~helpers';

const styles = StyleSheet.create({
  image: {
    padding: scaleSpace(12),
    width: scaleSpace(20),
    height: scaleSpace(20),
  },
});

export default styles;
