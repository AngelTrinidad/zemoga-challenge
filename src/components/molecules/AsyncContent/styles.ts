import {StyleSheet} from 'react-native';

import {scaleSpace} from '~helpers';

const styles = StyleSheet.create({
  errorMessage: {
    marginBottom: scaleSpace(12),
    fontWeight: '800',
  },
});

export default styles;
