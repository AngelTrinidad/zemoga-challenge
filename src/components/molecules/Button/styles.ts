import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  buttonText: {
    color: 'white',
  },
});

export default styles;
