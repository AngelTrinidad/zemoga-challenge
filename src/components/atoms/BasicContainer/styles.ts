import {StyleSheet} from 'react-native';

import {DEFAULT_SPACES} from '~constants/styles';

const styles = StyleSheet.create({
  safeArea: {
    flex: 1,
  },
  content: {
    padding: DEFAULT_SPACES.paddingScreen,
  },
});

export default styles;
