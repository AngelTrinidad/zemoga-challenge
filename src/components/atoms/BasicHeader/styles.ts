import {StyleSheet} from 'react-native';

import {DEFAULT_SPACES} from '~constants/styles';
import {scaleSpace} from '~helpers';

const styles = StyleSheet.create({
  header: {
    paddingHorizontal: DEFAULT_SPACES.paddingScreen,
    paddingVertical: scaleSpace(24),
  },
});

export default styles;
