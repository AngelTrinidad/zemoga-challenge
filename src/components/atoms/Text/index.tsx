import React, {FC} from 'react';
import {Text as RNText, StyleSheet, TextProps} from 'react-native';

import {DEFAULT_FONT} from '~constants/styles';
import {scaleText} from '~helpers';

/**
 * Conver font weight to the corresponding Font Family
 * @param value Font Weight value
 * @returns Font Family
 */
const getFontByWeight = (value: string | undefined) => {
  switch (value) {
    case '400':
    default:
      return DEFAULT_FONT.regular;
    case '800':
      return DEFAULT_FONT.bold;
  }
};

interface Props extends TextProps {}

const Text: FC<Props> = ({style, children, ...props}) => {
  const flattenStyles = StyleSheet.flatten(style);

  // Flattening the styles and defining scale font size and font family
  const textStyles = {
    ...flattenStyles,
    ...(!!flattenStyles && {
      fontSize: scaleText(flattenStyles.fontSize || 14),
      fontFamily: getFontByWeight(flattenStyles.fontWeight),
    }),
  };

  return (
    <RNText style={textStyles} {...props}>
      {children}
    </RNText>
  );
};

export default Text;
