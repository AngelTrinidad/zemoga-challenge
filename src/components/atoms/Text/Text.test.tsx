import React from 'react';

import {render} from '@testing-library/react-native';

import Text from '.';

import {DEFAULT_FONT} from '~constants/styles';

describe('<BasicButton />', () => {
  let component: any;

  beforeEach(() => {
    component = render(<Text>Hello</Text>);
  });

  it('should renders correctly', () => {
    expect(component).toBeDefined();
  });

  it('should has the same snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should convert font weight to font family', () => {
    const TextComponent = render(
      <Text style={{fontWeight: '800'}}>Hello</Text>,
    );
    const jsonComponent = TextComponent.toJSON() as any;
    expect(jsonComponent.props.style.fontFamily).toBe(DEFAULT_FONT.bold);
  });
});
