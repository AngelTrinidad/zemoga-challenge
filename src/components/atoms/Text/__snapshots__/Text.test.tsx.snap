// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasicButton /> should has the same snapshot 1`] = `
<Text
  style={{}}
>
  Hello
</Text>
`;
